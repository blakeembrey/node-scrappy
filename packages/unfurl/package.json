{
  "name": "@borderless/unfurl",
  "version": "0.6.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "Extract rich metadata from URLs",
  "license": "Apache-2.0",
  "repository": {
    "type": "git",
    "url": "git://github.com/borderless/unfurl.git"
  },
  "author": {
    "name": "Blake Embrey",
    "email": "hello@blakeembrey.com",
    "url": "http://blakeembrey.me"
  },
  "homepage": "https://github.com/borderless/unfurl",
  "bugs": {
    "url": "https://github.com/borderless/unfurl/issues"
  },
  "main": "dist/index.js",
  "scripts": {
    "format": "ts-scripts format",
    "lint": "ts-scripts lint",
    "prepare": "ts-scripts build",
    "specs": "ts-scripts specs",
    "test": "ts-scripts test"
  },
  "files": [
    "dist/"
  ],
  "keywords": [
    "unfurl",
    "metadata",
    "scraper",
    "rich",
    "snippets",
    "info",
    "rdfa",
    "open graph",
    "oembed",
    "microdata",
    "json-ld"
  ]
}
