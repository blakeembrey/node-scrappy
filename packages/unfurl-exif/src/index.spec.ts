import { FIXTURE_URLS, request } from "test";
import { urlScraper } from "@borderless/unfurl";
import exif from "./index";

const scrape = urlScraper({ request, plugins: [exif] });

describe("unfurl-exif", () => {
  for (const url of FIXTURE_URLS) {
    it(`should read ${url}`, async () => {
      const result = await scrape(url);
      expect(result).toMatchSnapshot();
    });
  }
});
