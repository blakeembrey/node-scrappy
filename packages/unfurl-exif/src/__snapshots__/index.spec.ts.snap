// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unfurl-exif should read http://cloudinary.com/pricing 1`] = `
Object {
  "type": "link",
  "url": "https://cloudinary.com/pricing",
}
`;

exports[`unfurl-exif should read http://cnn.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.cnn.com/",
}
`;

exports[`unfurl-exif should read http://d.pr/a/q3z9 1`] = `
Object {
  "type": "link",
  "url": "https://d.pr/a/q3z9",
}
`;

exports[`unfurl-exif should read http://dealmoon.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.dealmoon.com/",
}
`;

exports[`unfurl-exif should read http://everlane.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.everlane.com/",
}
`;

exports[`unfurl-exif should read http://franz.com/agraph/allegrograph/ 1`] = `
Object {
  "type": "link",
  "url": "https://franz.com:443/agraph/allegrograph/",
}
`;

exports[`unfurl-exif should read http://gizmodo.com/the-dnc-hack-was-much-bigger-than-we-thought-1785145268?utm_campaign=socialflow_gizmodo_twitter&utm_source=gizmodo_twitter&utm_medium=socialflow 1`] = `
Object {
  "type": "link",
  "url": "https://gizmodo.com/the-dnc-hack-was-much-bigger-than-we-thought-1785145268?utm_campaign=socialflow_gizmodo_twitter&utm_source=gizmodo_twitter&utm_medium=socialflow",
}
`;

exports[`unfurl-exif should read http://hackerne.ws 1`] = `
Object {
  "type": "link",
  "url": "https://news.ycombinator.com/",
}
`;

exports[`unfurl-exif should read http://html5doctor.com/microdata/ 1`] = `
Object {
  "type": "link",
  "url": "http://html5doctor.com/microdata/",
}
`;

exports[`unfurl-exif should read http://html5doctor.com/wp-content/uploads/2011/08/rich-snippet-data2.png 1`] = `
Object {
  "camera": Object {
    "lensMake": undefined,
    "lensModel": undefined,
    "make": undefined,
    "model": undefined,
    "orientation": undefined,
    "software": undefined,
  },
  "dateCreated": undefined,
  "dateModified": undefined,
  "encodingFormat": "image/png",
  "height": 341,
  "type": "image",
  "url": "http://html5doctor.com/wp-content/uploads/2011/08/rich-snippet-data2.png",
  "width": 495,
}
`;

exports[`unfurl-exif should read http://ideas.ted.com/why-i-learned-20-languages-and-what-i-learned-about-myself-in-the-process/?utm_campaign=social&utm_medium=referral&utm_source=facebook.com&utm_content=ideas-blog&utm_term=education 1`] = `
Object {
  "type": "link",
  "url": "https://ideas.ted.com/why-i-learned-20-languages-and-what-i-learned-about-myself-in-the-process/?utm_campaign=social&utm_medium=referral&utm_source=facebook.com&utm_content=ideas-blog&utm_term=education",
}
`;

exports[`unfurl-exif should read http://knexjs.org/ 1`] = `
Object {
  "type": "link",
  "url": "http://knexjs.org/",
}
`;

exports[`unfurl-exif should read http://mashable.com/2016/06/17/battle-of-the-bastards-game-of-thrones-fantasy-ending/#L2pP7_P5k5q3 1`] = `
Object {
  "type": "link",
  "url": "https://mashable.com/2016/06/17/battle-of-the-bastards-game-of-thrones-fantasy-ending/",
}
`;

exports[`unfurl-exif should read http://ogp.me/ 1`] = `
Object {
  "type": "link",
  "url": "https://ogp.me/",
}
`;

exports[`unfurl-exif should read http://shop.boostedboards.com/products/boosted-dual-plus?variant=1141824744 1`] = `
Object {
  "type": "link",
  "url": "https://buy.boostedboards.com/vehicles/all-boards",
}
`;

exports[`unfurl-exif should read http://store.steampowered.com/app/8930/ 1`] = `
Object {
  "type": "link",
  "url": "https://store.steampowered.com/app/8930/",
}
`;

exports[`unfurl-exif should read http://waitbutwhy.com/2016/09/marriage-decision.html 1`] = `
Object {
  "type": "link",
  "url": "https://waitbutwhy.com/2016/09/marriage-decision.html",
}
`;

exports[`unfurl-exif should read http://web.cs.ucdavis.edu/~rogaway/papers/moral-fn.pdf 1`] = `
Object {
  "type": "link",
  "url": "https://web.cs.ucdavis.edu/~rogaway/papers/moral-fn.pdf",
}
`;

exports[`unfurl-exif should read http://www.ebay.com/itm/Outdoor-Wicker-Patio-Furniture-Sofa-3-Seater-Luxury-Comfort-Brown-Wicker-Couch-/381228738769?&_trksid=p2056016.l4276 1`] = `
Object {
  "type": "link",
  "url": "https://www.ebay.com/p/2164370744?iid=381228738769",
}
`;

exports[`unfurl-exif should read http://www.imdb.com/title/tt1375666/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.imdb.com/title/tt1375666/",
}
`;

exports[`unfurl-exif should read http://www.lifehacker.com.au/2016/08/some-bill-providers-automatically-update-your-credit-card-when-you-get-a-new-one/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.lifehacker.com.au/2016/08/some-bill-providers-automatically-update-your-credit-card-when-you-get-a-new-one/",
}
`;

exports[`unfurl-exif should read http://www.mysteryranch.com/asap-pack 1`] = `
Object {
  "type": "link",
  "url": "https://www.mysteryranch.com/asap-pack",
}
`;

exports[`unfurl-exif should read http://www.newegg.com/Product/Product.aspx?Item=28-840-014 1`] = `
Object {
  "type": "link",
  "url": "https://www.newegg.com/xyzprinting-da-vinci-2-0-duo/p/N82E16828840014",
}
`;

exports[`unfurl-exif should read http://www.news.com.au/world/breaking-news/uk-man-tried-to-kill-trump-court-papers/news-story/c4116603f54f1b7c88339cd039c7e123 1`] = `
Object {
  "type": "link",
  "url": "https://www.news.com.au/404",
}
`;

exports[`unfurl-exif should read http://www.nytimes.com/2016/06/15/opinion/campaign-stops/decoding-donald-trump.html?action=click&module=MostEmailed&region=Lists&pgtype=collection 1`] = `
Object {
  "type": "link",
  "url": "https://www.nytimes.com/2016/06/15/opinion/campaign-stops/decoding-donald-trump.html?action=click&module=MostEmailed&region=Lists&pgtype=collection",
}
`;

exports[`unfurl-exif should read http://www.reddit.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.reddit.com/",
}
`;

exports[`unfurl-exif should read http://www.theverge.com/2016/8/10/12416766/google-white-house-ostp-emails?utm_campaign=theverge&utm_content=chorus&utm_medium=social&utm_source=twitter 1`] = `
Object {
  "type": "link",
  "url": "https://www.theverge.com/2016/8/10/12416766/google-white-house-ostp-emails?utm_campaign=theverge&utm_content=chorus&utm_medium=social&utm_source=twitter",
}
`;

exports[`unfurl-exif should read http://www.yelp.com/biz/boba-guys-san-francisco-4 1`] = `
Object {
  "type": "link",
  "url": "https://www.yelp.com/biz/boba-guys-san-francisco-4",
}
`;

exports[`unfurl-exif should read http://xkcd.com/208/ 1`] = `
Object {
  "type": "link",
  "url": "https://xkcd.com/208/",
}
`;

exports[`unfurl-exif should read https://aaronparecki.com/2016/08/23/2/micropub-cr?utm_medium=email&utm_source=html5weekly 1`] = `
Object {
  "type": "link",
  "url": "https://aaronparecki.com/2016/08/23/2/micropub-cr?utm_medium=email&utm_source=html5weekly",
}
`;

exports[`unfurl-exif should read https://adactio.com/journal/9881 1`] = `
Object {
  "type": "link",
  "url": "https://adactio.com/journal/9881",
}
`;

exports[`unfurl-exif should read https://atom.io/themes/aesthetic-ui 1`] = `
Object {
  "type": "link",
  "url": "https://atom.io/themes/aesthetic-ui",
}
`;

exports[`unfurl-exif should read https://bjango.com/articles/pngoptimisation/ 1`] = `
Object {
  "type": "link",
  "url": "https://bjango.com/articles/pngoptimisation/",
}
`;

exports[`unfurl-exif should read https://bjango.com/mac/istatmenus/ 1`] = `
Object {
  "type": "link",
  "url": "https://bjango.com/mac/istatmenus/",
}
`;

exports[`unfurl-exif should read https://c1.staticflickr.com/9/8699/28229743586_cd32cea242_o.jpg 1`] = `
Object {
  "camera": Object {
    "lensMake": undefined,
    "lensModel": "17-70mm",
    "make": "Canon",
    "model": "Canon EOS 700D",
    "orientation": "top-left",
    "software": "Adobe Photoshop Lightroom 6.1 (Windows)",
  },
  "dateCreated": 2016-07-11T21:33:20.071Z,
  "dateModified": 2016-07-11T23:35:24.000Z,
  "encodingFormat": "image/jpeg",
  "height": 5110,
  "type": "image",
  "url": "https://c1.staticflickr.com/9/8699/28229743586_cd32cea242_o.jpg",
  "width": 3407,
}
`;

exports[`unfurl-exif should read https://c2.staticflickr.com/4/3930/15428630771_61e9dcf6f4_o.jpg 1`] = `
Object {
  "camera": Object {
    "lensMake": "Apple",
    "lensModel": "iPhone 5 back camera 4.12mm f/2.4",
    "make": undefined,
    "model": undefined,
    "orientation": "top-left",
    "software": "Instagram",
  },
  "dateCreated": undefined,
  "dateModified": undefined,
  "encodingFormat": "image/jpeg",
  "height": 2448,
  "type": "image",
  "url": "https://c2.staticflickr.com/4/3930/15428630771_61e9dcf6f4_o.jpg",
  "width": 2448,
}
`;

exports[`unfurl-exif should read https://cansecwest.com/slides/2015/Liang_CanSecWest2015.pdf 1`] = `
Object {
  "type": "link",
  "url": "https://cansecwest.com/slides/2015/Liang_CanSecWest2015.pdf",
}
`;

exports[`unfurl-exif should read https://dev.twitter.com/cards/types/player 1`] = `
Object {
  "type": "link",
  "url": "https://developer.twitter.com/en/docs/tweets/optimize-with-cards/overview/player-card",
}
`;

exports[`unfurl-exif should read https://developers.google.com/search/docs/guides/intro-structured-data 1`] = `
Object {
  "type": "link",
  "url": "https://developers.google.com/search/docs/guides/intro-structured-data",
}
`;

exports[`unfurl-exif should read https://discourse.codinghorror.com/t/the-raspberry-pi-has-revolutionized-emulation/4462/29 1`] = `
Object {
  "type": "link",
  "url": "https://discourse.codinghorror.com/t/the-raspberry-pi-has-revolutionized-emulation/4462/29",
}
`;

exports[`unfurl-exif should read https://docs.google.com/document/d/1GnsFxQZWERvB5A2cYnmpmNzgH_zAtUsUMQ-th1em2jQ/edit?usp=sharing 1`] = `
Object {
  "type": "link",
  "url": "https://docs.google.com/document/d/1GnsFxQZWERvB5A2cYnmpmNzgH_zAtUsUMQ-th1em2jQ/edit?usp=sharing",
}
`;

exports[`unfurl-exif should read https://docs.google.com/spreadsheets/d/1teKblpByMmLaSmRAqDCyLXf0RApcJg3sg4E0MMmfrPg/edit?usp=sharing 1`] = `
Object {
  "type": "link",
  "url": "https://docs.google.com/spreadsheets/d/1teKblpByMmLaSmRAqDCyLXf0RApcJg3sg4E0MMmfrPg/edit?usp=sharing",
}
`;

exports[`unfurl-exif should read https://drive.google.com/file/d/0B59Tysg-nEQZOGhsU0U5QXo0Sjg/view?usp=sharing 1`] = `
Object {
  "type": "link",
  "url": "https://drive.google.com/file/d/0B59Tysg-nEQZOGhsU0U5QXo0Sjg/view?usp=sharing",
}
`;

exports[`unfurl-exif should read https://droplr.com/droplr-addition-google-chrome-extension 1`] = `
Object {
  "type": "link",
  "url": "https://droplr.com/droplr-addition-google-chrome-extension",
}
`;

exports[`unfurl-exif should read https://en.wikipedia.org/wiki/Jean-Luc_Picard 1`] = `
Object {
  "type": "link",
  "url": "https://en.wikipedia.org/wiki/Jean-Luc_Picard",
}
`;

exports[`unfurl-exif should read https://en.wikipedia.org/wiki/TypeScript 1`] = `
Object {
  "type": "link",
  "url": "https://en.wikipedia.org/wiki/TypeScript",
}
`;

exports[`unfurl-exif should read https://foursquare.com/v/boba-guys/51abb0a3498eb42c0d5cf324 1`] = `
Object {
  "type": "link",
  "url": "https://foursquare.com/v/boba-guys/51abb0a3498eb42c0d5cf324",
}
`;

exports[`unfurl-exif should read https://foursquare.com/v/bondi-beach/4b058763f964a520848f22e3 1`] = `
Object {
  "type": "link",
  "url": "https://foursquare.com/v/bondi-beach/4b058763f964a520848f22e3",
}
`;

exports[`unfurl-exif should read https://foursquare.com/v/bourke-street-bakery/4b0b4a63f964a520fa2f23e3 1`] = `
Object {
  "type": "link",
  "url": "https://foursquare.com/v/bourke-street-bakery/4b0b4a63f964a520fa2f23e3",
}
`;

exports[`unfurl-exif should read https://foursquare.com/v/darling-harbour/4b058762f964a5201d8f22e3 1`] = `
Object {
  "type": "link",
  "url": "https://foursquare.com/v/darling-harbour/4b058762f964a5201d8f22e3",
}
`;

exports[`unfurl-exif should read https://foursquare.com/v/mrs-macquaries-point/4b3c2445f964a520348225e3 1`] = `
Object {
  "type": "link",
  "url": "https://foursquare.com/v/mrs-macquaries-point/4b3c2445f964a520348225e3",
}
`;

exports[`unfurl-exif should read https://foursquare.com/v/royal-botanic-garden/4b058761f964a520188f22e3 1`] = `
Object {
  "type": "link",
  "url": "https://foursquare.com/v/royal-botanic-garden/4b058761f964a520188f22e3",
}
`;

exports[`unfurl-exif should read https://foursquare.com/v/sydney-opera-house/4b058762f964a5201b8f22e3 1`] = `
Object {
  "type": "link",
  "url": "https://foursquare.com/v/sydney-opera-house/4b058762f964a5201b8f22e3",
}
`;

exports[`unfurl-exif should read https://foursquare.com/v/the-baxter-inn/4ed4896c775b45f6ed7b0182 1`] = `
Object {
  "type": "link",
  "url": "https://foursquare.com/v/the-baxter-inn/4ed4896c775b45f6ed7b0182",
}
`;

exports[`unfurl-exif should read https://github.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://github.com/",
}
`;

exports[`unfurl-exif should read https://github.com/Microsoft/TypeScript/issues/10462 1`] = `
Object {
  "type": "link",
  "url": "https://github.com/Microsoft/TypeScript/issues/10462",
}
`;

exports[`unfurl-exif should read https://github.com/blakeembrey 1`] = `
Object {
  "type": "link",
  "url": "https://github.com/blakeembrey",
}
`;

exports[`unfurl-exif should read https://github.com/blakeembrey/popsicle 1`] = `
Object {
  "type": "link",
  "url": "https://github.com/serviejs/popsicle",
}
`;

exports[`unfurl-exif should read https://github.com/blog/2242-git-2-10-has-been-released 1`] = `
Object {
  "type": "link",
  "url": "https://github.blog/2016-09-03-git-2-10-has-been-released/",
}
`;

exports[`unfurl-exif should read https://github.com/moment/moment/pull/3323 1`] = `
Object {
  "type": "link",
  "url": "https://github.com/moment/moment/pull/3323",
}
`;

exports[`unfurl-exif should read https://i.imgur.com/mvUPRyV.png 1`] = `
Object {
  "camera": Object {
    "lensMake": undefined,
    "lensModel": undefined,
    "make": undefined,
    "model": undefined,
    "orientation": undefined,
    "software": undefined,
  },
  "dateCreated": undefined,
  "dateModified": undefined,
  "encodingFormat": "image/png",
  "height": 484,
  "type": "image",
  "url": "https://i.imgur.com/mvUPRyV.png",
  "width": 773,
}
`;

exports[`unfurl-exif should read https://ianstormtaylor.com/design-tip-never-use-black/ 1`] = `
Object {
  "type": "link",
  "url": "https://ianstormtaylor.com/design-tip-never-use-black/",
}
`;

exports[`unfurl-exif should read https://itunes.apple.com/us/app/pokemon-go/id1094591345?mt=8 1`] = `
Object {
  "type": "link",
  "url": "https://apps.apple.com/us/app/pokemon-go/id1094591345",
}
`;

exports[`unfurl-exif should read https://itunes.apple.com/us/movie/the-avengers/id533654020 1`] = `
Object {
  "type": "link",
  "url": "https://itunes.apple.com/us/movie/the-avengers/id533654020",
}
`;

exports[`unfurl-exif should read https://learnxinyminutes.com/docs/standard-ml/ 1`] = `
Object {
  "type": "link",
  "url": "https://learnxinyminutes.com/docs/standard-ml/",
}
`;

exports[`unfurl-exif should read https://medium.com/slack-developer-blog/everything-you-ever-wanted-to-know-about-unfurling-but-were-afraid-to-ask-or-how-to-make-your-e64b4bb9254#.a0wjf4ltt 1`] = `
Object {
  "type": "link",
  "url": "https://medium.com/slack-developer-blog/everything-you-ever-wanted-to-know-about-unfurling-but-were-afraid-to-ask-or-how-to-make-your-e64b4bb9254#.a0wjf4ltt",
}
`;

exports[`unfurl-exif should read https://moz.com/community/q/are-rel-author-and-rel-publisher-meta-tags-currently-in-use 1`] = `
Object {
  "type": "link",
  "url": "https://moz.com/community/q/are-rel-author-and-rel-publisher-meta-tags-currently-in-use",
}
`;

exports[`unfurl-exif should read https://news.ycombinator.com/item?id=12282756 1`] = `
Object {
  "type": "link",
  "url": "https://news.ycombinator.com/item?id=12282756",
}
`;

exports[`unfurl-exif should read https://pragprog.com/book/btlang/seven-languages-in-seven-weeks 1`] = `
Object {
  "type": "link",
  "url": "https://pragprog.com/book/btlang/seven-languages-in-seven-weeks",
}
`;

exports[`unfurl-exif should read https://schema.org/WebSite 1`] = `
Object {
  "type": "link",
  "url": "https://schema.org/WebSite",
}
`;

exports[`unfurl-exif should read https://schema.org/docs/schema_org_rdfa.html 1`] = `
Object {
  "type": "link",
  "url": "https://schema.org/docs/schema_org_rdfa.html",
}
`;

exports[`unfurl-exif should read https://segment.com/blog/the-segment-aws-stack/ 1`] = `
Object {
  "type": "link",
  "url": "https://segment.com/blog/the-segment-aws-stack/",
}
`;

exports[`unfurl-exif should read https://slant6.bandcamp.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://slant6.bandcamp.com/",
}
`;

exports[`unfurl-exif should read https://smile.amazon.com/gp/product/1937785734/ref=s9_qpp_gw_d99_g14_i5_r?ie=UTF8&fpl=fresh&pf_rd_m=ATVPDKIKX0DER&pf_rd_s=desktop-1&pf_rd_r=92JG667YC8526036ANPR&pf_rd_t=36701&pf_rd_p=6aad23bd-3035-4a40-b691-0eefb1a18396&pf_rd_i=desktop 1`] = `
Object {
  "type": "link",
  "url": "https://www.amazon.com/gp/product/1937785734/ref=s9_qpp_gw_d99_g14_i5_r?ie=UTF8&fpl=fresh&pf_rd_m=ATVPDKIKX0DER&pf_rd_s=desktop-1&pf_rd_r=92JG667YC8526036ANPR&pf_rd_t=36701&pf_rd_p=6aad23bd-3035-4a40-b691-0eefb1a18396&pf_rd_i=desktop&pldnSite=1",
}
`;

exports[`unfurl-exif should read https://soundcloud.com/lifeofdesiigner/desiigner-panda 1`] = `
Object {
  "type": "link",
  "url": "https://soundcloud.com/lifeofdesiigner/desiigner-panda",
}
`;

exports[`unfurl-exif should read https://soundcloud.com/tobiasvanschneider/ntmy-episode-1-pieter-levels 1`] = `
Object {
  "type": "link",
  "url": "https://soundcloud.com/tobiasvanschneider/ntmy-episode-1-pieter-levels",
}
`;

exports[`unfurl-exif should read https://spreadprivacy.com/privacy-risks-note-apps/ 1`] = `
Object {
  "type": "link",
  "url": "https://spreadprivacy.com/privacy-risks-note-apps/",
}
`;

exports[`unfurl-exif should read https://stratechery.com/2016/google-uber-and-the-evolution-of-transportation-as-a-service/ 1`] = `
Object {
  "type": "link",
  "url": "https://stratechery.com/2016/google-uber-and-the-evolution-of-transportation-as-a-service/",
}
`;

exports[`unfurl-exif should read https://tailwindui.com/pricing 1`] = `
Object {
  "type": "link",
  "url": "https://tailwindui.com/pricing",
}
`;

exports[`unfurl-exif should read https://techcrunch.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://techcrunch.com/",
}
`;

exports[`unfurl-exif should read https://techcrunch.com/2016/06/17/the-europas-awards-2016-honored-the-best-tech-startups-in-europe/ 1`] = `
Object {
  "type": "link",
  "url": "https://techcrunch.com/2016/06/17/the-europas-awards-2016-honored-the-best-tech-startups-in-europe/",
}
`;

exports[`unfurl-exif should read https://techcrunch.com/2020/02/23/sensors-are-the-next-big-thing-in-space-not-starships/ 1`] = `
Object {
  "type": "link",
  "url": "https://techcrunch.com/2020/02/23/sensors-are-the-next-big-thing-in-space-not-starships/",
}
`;

exports[`unfurl-exif should read https://trakt.tv/movies/kung-fury-2015 1`] = `
Object {
  "type": "link",
  "url": "https://trakt.tv/movies/kung-fury-2015",
}
`;

exports[`unfurl-exif should read https://twitter.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://twitter.com/",
}
`;

exports[`unfurl-exif should read https://twitter.com/SeanTAllen/status/764993929469161472 1`] = `
Object {
  "type": "link",
  "url": "https://twitter.com/SeanTAllen/status/764993929469161472",
}
`;

exports[`unfurl-exif should read https://twitter.com/alexisohanian/status/764997551384776704 1`] = `
Object {
  "type": "link",
  "url": "https://twitter.com/alexisohanian/status/764997551384776704",
}
`;

exports[`unfurl-exif should read https://twitter.com/blakeembrey 1`] = `
Object {
  "type": "link",
  "url": "https://twitter.com/blakeembrey",
}
`;

exports[`unfurl-exif should read https://twitter.com/ericclemmons/status/749223563790471169 1`] = `
Object {
  "type": "link",
  "url": "https://twitter.com/ericclemmons/status/749223563790471169",
}
`;

exports[`unfurl-exif should read https://twitter.com/typescriptlang/status/743113612407889920 1`] = `
Object {
  "type": "link",
  "url": "https://twitter.com/typescript/status/743113612407889920",
}
`;

exports[`unfurl-exif should read https://upload.wikimedia.org/wikipedia/commons/a/a1/Johnrogershousemay2020.webp 1`] = `
Object {
  "type": "image",
  "url": "https://upload.wikimedia.org/wikipedia/commons/a/a1/Johnrogershousemay2020.webp",
}
`;

exports[`unfurl-exif should read https://web.eecs.umich.edu/~mihalcea/papers/mihalcea.emnlp04.pdf 1`] = `
Object {
  "type": "link",
  "url": "https://web.eecs.umich.edu/~mihalcea/papers/mihalcea.emnlp04.pdf",
}
`;

exports[`unfurl-exif should read https://www.airbnb.com/rooms/2250401?checkin=06%2F24%2F2016&checkout=06%2F30%2F2016&guests=1&s=eI3nl9s6 1`] = `
Object {
  "type": "link",
  "url": "https://www.airbnb.com/rooms/2250401?checkin=06%2F24%2F2016&checkout=06%2F30%2F2016&guests=1&s=eI3nl9s6",
}
`;

exports[`unfurl-exif should read https://www.battlefield.com/buy/battlefield-1 1`] = `
Object {
  "type": "link",
  "url": "https://www.battlefield.com/buy/battlefield-1",
}
`;

exports[`unfurl-exif should read https://www.coursera.org/learn/machine-learning 1`] = `
Object {
  "type": "link",
  "url": "https://www.coursera.org/learn/machine-learning",
}
`;

exports[`unfurl-exif should read https://www.etsy.com/listing/549504785/pokenatomy-the-science-of-pokemon-an?ref=hp_prn&frs=1 1`] = `
Object {
  "type": "link",
  "url": "https://www.etsy.com/listing/549504785/pokenatomy-the-science-of-pokemon-an?ref=hp_prn&frs=1",
}
`;

exports[`unfurl-exif should read https://www.facebook.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.facebook.com/",
}
`;

exports[`unfurl-exif should read https://www.flickr.com/photos/fabianf_/28229743586/sizes/o/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.flickr.com/photos/fabianf_/28229743586/sizes/l/",
}
`;

exports[`unfurl-exif should read https://www.flickr.com/photos/timdrivas/27999498362/in/explore-2016-07-05/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.flickr.com/photos/timdrivas/27999498362/in/explore-2016-07-05/",
}
`;

exports[`unfurl-exif should read https://www.google.com/maps/place/Boba+Guys/@37.7600441,-122.4233333,17z/data=!4m5!3m4!1s0x808f7e3cfdb89265:0x8ae0820c41111f70!8m2!3d37.7600017!4d-122.4211124 1`] = `
Object {
  "type": "link",
  "url": "https://www.google.com/maps/place/Boba+Guys/@37.7600441,-122.4233333,17z/data=!4m5!3m4!1s0x808f7e3cfdb89265:0x8ae0820c41111f70!8m2!3d37.7600017!4d-122.4211124",
}
`;

exports[`unfurl-exif should read https://www.gstatic.com/webp/gallery/1.webp 1`] = `
Object {
  "type": "image",
  "url": "https://www.gstatic.com/webp/gallery/1.webp",
}
`;

exports[`unfurl-exif should read https://www.instagram.com/p/BG0m4IDGaqk/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.instagram.com/p/BG0m4IDGaqk/",
}
`;

exports[`unfurl-exif should read https://www.khanacademy.org/economics-finance-domain/core-finance/stock-and-bonds/stocks-intro-tutorial/v/bonds-vs-stocks 1`] = `
Object {
  "type": "link",
  "url": "https://www.khanacademy.org/economics-finance-domain/core-finance/stock-and-bonds/stocks-intro-tutorial/v/bonds-vs-stocks",
}
`;

exports[`unfurl-exif should read https://www.kickstarter.com/projects/lactate-threshold/lvl-the-first-wearable-hydration-monitor?ref=category_featured 1`] = `
Object {
  "type": "link",
  "url": "https://www.kickstarter.com/projects/lactate-threshold/lvl-the-first-wearable-hydration-monitor?ref=category_featured",
}
`;

exports[`unfurl-exif should read https://www.mulesoft.com/contact 1`] = `
Object {
  "type": "link",
  "url": "https://www.mulesoft.com/contact",
}
`;

exports[`unfurl-exif should read https://www.npmjs.com/package/filenamify 1`] = `
Object {
  "type": "link",
  "url": "https://www.npmjs.com/package/filenamify",
}
`;

exports[`unfurl-exif should read https://www.opendoor.com/homes/21-via-hacienda-rancho-santa-margarita-ca-92688 1`] = `
Object {
  "type": "link",
  "url": "https://www.opendoor.com/homes/21-via-hacienda-rancho-santa-margarita-ca-92688",
}
`;

exports[`unfurl-exif should read https://www.quora.com/How-do-I-impress-a-computer-programmer-on-a-date 1`] = `
Object {
  "type": "link",
  "url": "https://www.quora.com/How-do-I-impress-a-computer-programmer-on-a-date",
}
`;

exports[`unfurl-exif should read https://www.reddit.com/r/news/comments/4p1enj/uk_man_tried_to_kill_trump_court_papers/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.reddit.com/r/news/comments/4p1enj/uk_man_tried_to_kill_trump_court_papers/",
}
`;

exports[`unfurl-exif should read https://www.redfin.com/CA/Oakland/2325-Valley-St-94612/unit-201/home/173701090 1`] = `
Object {
  "type": "link",
  "url": "https://www.redfin.com/CA/Oakland/2325-Valley-St-94612/unit-201/home/173701090",
}
`;

exports[`unfurl-exif should read https://www.rottentomatoes.com/m/inception/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.rottentomatoes.com/m/inception/",
}
`;

exports[`unfurl-exif should read https://www.sciencedaily.com/releases/2020/03/200311121832.htm 1`] = `
Object {
  "type": "link",
  "url": "https://www.sciencedaily.com/releases/2020/03/200311121832.htm",
}
`;

exports[`unfurl-exif should read https://www.spotify.com/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.spotify.com/us/",
}
`;

exports[`unfurl-exif should read https://www.ted.com/playlists/321/talks_to_form_better_habits?utm_campaign=social&utm_medium=referral&utm_source=facebook.com&utm_content=playlist&utm_term=social-science 1`] = `
Object {
  "type": "link",
  "url": "https://www.ted.com/playlists/321/talks_to_form_better_habits?utm_campaign=social&utm_medium=referral&utm_source=facebook.com&utm_content=playlist&utm_term=social-science",
}
`;

exports[`unfurl-exif should read https://www.ted.com/talks/tim_harford_how_messy_problems_can_inspire_creativity 1`] = `
Object {
  "type": "link",
  "url": "https://www.ted.com/talks/tim_harford_how_frustration_can_make_us_more_creative",
}
`;

exports[`unfurl-exif should read https://www.w3.org/TR/json-ld/ 1`] = `
Object {
  "type": "link",
  "url": "https://www.w3.org/TR/json-ld/",
}
`;

exports[`unfurl-exif should read https://www.youtube.com/watch?v=W9ZnpIGvZUo 1`] = `
Object {
  "type": "link",
  "url": "https://www.youtube.com/watch?v=W9ZnpIGvZUo",
}
`;

exports[`unfurl-exif should read https://www.youtube.com/watch?v=ZynZdGqxT7Q 1`] = `
Object {
  "type": "link",
  "url": "https://www.youtube.com/watch?v=ZynZdGqxT7Q",
}
`;

exports[`unfurl-exif should read https://yoast.com/wordpress/plugins/seo/ 1`] = `
Object {
  "type": "link",
  "url": "https://yoast.com/wordpress/plugins/seo/",
}
`;
